### PR TITLE
YQ-2730: use docker-compose random port mapping in connector integration tests

### PR DESCRIPTION
**Pull request by vitalyisaev2**
* Add retries when making PG connections from connector integration tests
* Use Docker port randomization for data sources launched in containers

### CODE REVIEW

File: ydb/library/yql/providers/generic/connector/tests/utils/settings.py [hunk 1]
```diff
@@ -41,26 +44,29 @@ class PostgreSQL:
...
             connector=cls.Connector(
                 grpc_host='localhost',
-                grpc_port=50051,
+                grpc_port=endpoint_determiner.get_port('fq-connector-go', 50051),
                 paging_bytes_per_page=4 * 1024 * 1024,
                 paging_prefetch_queue_capacity=2,
             ),
             clickhouse=cls.ClickHouse(
                 cluster_name='clickhouse_integration_test',
                 host='localhost',
```
Comment (tsmax2004) [inline ydb/library/yql/providers/generic/connector/tests/utils/settings.py:59]: does connector have access to `localhost` without 
```
network_mode: host
```
in `docker-compose.yml`?

maybe we should use service name (`clickhouse`) as host and internal port (`8123` or `9000`) as port here?

**Review by tsmax2004 — CHANGES_REQUESTED** [review]

File: ydb/library/yql/providers/generic/connector/tests/utils/docker_compose.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import yatest.common
+
+
+class EndpointDeterminer:
+    docker_compose_bin: os.PathLike
+    docker_compose_yml: os.PathLike
+
+    def __init__(self, docker_compose_yml: os.PathLike):
+        self.docker_compose_bin = os.path.realpath(
```
Comment (tsmax2004) [inline ydb/library/yql/providers/generic/connector/tests/utils/docker_compose.py:12]: `os.path.realpath` is necessary?

File: ydb/library/yql/providers/generic/connector/tests/utils/docker_compose.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            yatest.common.build_path('library/recipes/docker_compose/bin/docker-compose')
+        )
+        self.docker_compose_yml = docker_compose_yml
+
+    def get_port(self, service_name: str, internal_port: int) -> int:
+        cmd = ['docker-compose', '-f', self.docker_compose_yml, 'port', service_name, str(internal_port)]
```
Comment (tsmax2004) [inline ydb/library/yql/providers/generic/connector/tests/utils/docker_compose.py:18]: `self.docker_compose_bin` instead of `'docker-compose'`

**Review by tsmax2004 — CHANGES_REQUESTED** [review]

File: ydb/library/yql/providers/generic/connector/tests/utils/docker_compose.py (hunk 2 above)
Comment (vitalyisaev2) [reply ydb/library/yql/providers/generic/connector/tests/utils/docker_compose.py]: fixed

**Review by tsmax2004 — APPROVED** [review]

**Review by tsmax2004 — APPROVED** [review]

**Review by uzhastik — APPROVED** [review]

**Pull request merged**